### PR TITLE
Add stack overflow detection example

### PR DESCRIPTION
**Pull request by bjoernQ**
This adds an example of how to detect stack overflows.

The example currently needs the unreleased HAL - after the next release we can bump versions and remove the patch section.

Additionally, there are a few more unrelated changes
- bump esp32c3-hal version
- use a specific commit of esp-wifi
- make sure the esp-wifi example won't get you into problems on the DevKit-Rust (which only offers Serial/JTAG connector)

### CODE REVIEW

**Comment by bjoernQ** [comment]
probably best to wait until the next HAL release since needing patches is somewhat annoying

**Review by SergioGasquez — APPROVED** [review]
LGTM! Thanks for working on this! Let's wait for the hal releases and we can merge it.

File: advanced/stack_clash_protection/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+[dependencies]
+hal             = { package = "esp32c3-hal", version = "0.11.0" }
+esp-backtrace   = { version = "0.7.0", features = ["esp32c3", "panic-handler", "exception-handler", "print-uart"] }
+esp-println     = { version = "0.5.0", features = ["esp32c3"] }
+critical-section = "1.1.1"
+
+[patch.crates-io]
+esp32c3-hal = { git = "https://github.com/esp-rs/esp-hal", rev = "422eb2118676d4c1c938fa6cb2b8211fb177c232" }
+esp-hal-common = { git = "https://github.com/esp-rs/esp-hal", rev = "422eb2118676d4c1c938fa6cb2b8211fb177c232" }
```
Comment (SergioGasquez) [inline advanced/stack_clash_protection/Cargo.toml:15]: New hal are now published:
```suggestion
hal             = { package = "esp32c3-hal", version = "0.12.0" }
esp-backtrace   = { version = "0.8.0", features = ["esp32c3", "panic-handler", "exception-handler", "print-uart"] }
esp-println     = { version = "0.6.0", features = ["esp32c3"] }
critical-section = "1.1.1"
```
I've also updated `backtrace` and `println`

**Review by SergioGasquez — COMMENTED** [review]
Since new hal are now published we can unlock this PR

**Comment by bjoernQ** [comment]
I totally screwed up in rebasing this. It also seems that on main the http client doesn't compile? It's complaining about the lock file
I will close this PR and create a new one

**Comment by SergioGasquez** [comment]
> I totally screwed up in rebasing this. It also seems that on main the http client doesn't compile? It's complaining about the lock file I will close this PR and create a new one

Just noticed that the http_client is not built in CI, I'll open an issue and fix that! Thanks for the heads up

**Comment by bjoernQ** [comment]
Closed in favor of #16

**Pull request closed without merging**